lib/location/Kconfig

```diff
@@ -0,0 +1,146 @@
+#
+# Copyright (c) 2021 Nordic Semiconductor ASA
+#
+# SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+#
+
+menuconfig LOCATION
+	bool "Location"
+	depends on NRF_MODEM_LIB
+	depends on LTE_LINK_CONTROL
+
+if LOCATION
+
+config LOCATION_METHOD_GNSS
+	bool "Allow GNSS to be used for obtaining the location"
+	default y
+
+config LOCATION_METHOD_CELLULAR
+	bool "Allow cellular positioning to be used for obtaining the location"
+	select MULTICELL_LOCATION
+	default y
+
+config LOCATION_METHOD_WIFI
+	bool "Allow Wi-Fi positioning to be used for obtaining the location"
+	depends on WIFI
+	select REST_CLIENT
+	select CJSON_LIB
+	select NET_MGMT
+	select NET_MGMT_EVENT
+	select NET_MGMT_EVENT_INFO
+	select NET_L2_WIFI_MGMT
+
+config LOCATION_METHODS_LIST_SIZE
+	int "Maximum number of location methods in configuration"
+	default 3
+	help
+	  Maximum number of location methods within location_config structure.
+
+if LOCATION_METHOD_GNSS
+config LOCATION_METHOD_GNSS_AGPS_EXTERNAL
+	bool "Requesting A-GPS data is handled by the application outside of Location library."
+	depends on NRF_CLOUD_AGPS
```

Comment:
Just adding a comment on some line to get conversation to track.

After rebasing from ncs/main, PR https://github.com/nrfconnect/sdk-nrf/pull/5986
causes compilation issues due to missing `CONFIG_NRF_CLOUD_MQTT` flagging.

```
/home/tor1/work/ncs/nrf/subsys/net/lib/nrf_cloud/./include/nrf_cloud_fsm.h:70:36: error: unknown type name 'nrf_cloud_cell_pos_response_t'
void nfsm_set_cell_pos_response_cb(nrf_cloud_cell_pos_response_t cb);
```
I'll create a PR to get this fixed.

Reply:
https://github.com/nrfconnect/sdk-nrf/pull/6071 created to handle this

Reply:
Merged now